### PR TITLE
BUG: mne.set_bipolar_reference function to account for bad channels

### LINKED ISSUE #11243: ENH: mne.set_bipolar_reference should not ignore bad channels
### Description of the problem

When setting the bipolar reference between one good and one bad channel, the final bipolar channel is marked as good.

Bad channels are marked as bad because they contain artefacts. The bipolar reference between a bad channel and a good channel can never yield a good channel as a result. Therefore, the resulting bipolar channels should be marked as bad by mne.

[Jupyter Notebook: Steps to reproduce](https://github.com/mne-tools/mne-python/files/9771538/bipolar_ref.pdf)

### Steps to reproduce

```Python
A jupyter notebook to reproduce the problem is uploaded as a pdf.
```

### Link to data

_No response_

### Expected results

Bipolar referencing works but the bipolar channel is automatically marked as bad.

### Actual results

Bipolar referencing works but the bipolar channel is marked as a good channel even though it is clearly a bad channel.

### Additional information

Platform:         macOS-12.6-arm64-arm-64bit
Python:           3.10.6 | packaged by conda-forge | (main, Aug 22 2022, 20:41:22) [Clang 13.0.1 ]
Executable:       /Users/moritzgerster/anaconda3/envs/bids_neu/bin/python
CPU:              arm: 10 cores
Memory:           16.0 GB

mne:              1.2.dev0
numpy:            1.23.2 {unknown linalg bindings (threadpoolctl module not found: No module named 'threadpoolctl')}
scipy:            1.9.0
matplotlib:       3.5.3 {backend=module://matplotlib_inline.backend_inline}

sklearn:          Not found
numba:            Not found
nibabel:          4.0.1
nilearn:          Not found
dipy:             Not found
openmeeg:         Not found
cupy:             Not found
pandas:           1.4.3
pyvista:          0.36.1 {OpenGL 4.1 Metal - 76.3 via Apple M1 Pro}
pyvistaqt:        0.9.0
ipyvtklink:       Not found
vtk:              9.1.0
qtpy:             2.2.0 {None=None}
ipympl:           Not found
pyqtgraph:        Not found
pooch:            v1.6.0

mne_bids:         0.10
mne_nirs:         Not found
mne_features:     Not found
mne_qt_browser:   Not found
mne_connectivity: Not found
mne_icalabel:     Not found

### PR DESCRIPTION
**Pull request by moritz-gerster**
#### Reference issue
Fixes #11243.

#### What does this implement/fix?
When setting the bipolar reference between one good and one bad channel, the final bipolar channel is currently marked as good.

Bad channels are marked as bad because they contain artifacts. The bipolar reference between a bad channel and a good channel can never yield a good channel as a result. Therefore, the resulting bipolar channels should be marked as bad by mne.

#### Scenarios
Let us assume we have anode = ['Fc5.'] and cathode=['Fc3.'] and we want to create the bipolar channel ['Fc5.-Fc3.']. However, in our example ['Fc5.'] is marked as a bad channel. Currently, the resulting bipolar channel ['Fc5.-Fc3.'] is returned as a good channel. With our fix:

#### Scenarios
1. `on_bad="warn"` (default):
`set_bipolar_reference(raw, anode, cathode, ch_name=bipolar_chname, on_bad="warn")`
RuntimeWarning: Bipolar channels are based on bad channels: ['Fc5.-Fc3.'].

2. `on_bad="raise"`:
`set_bipolar_reference(raw, anode, cathode, ch_name=bipolar_chname, on_bad="raise")`
ValueError: Bipolar channels are based on bad channels: ['Fc5.-Fc3.'].

3. `on_bad="ignore"`:
`set_bipolar_reference(raw, anode, cathode, ch_name=bipolar_chname, on_bad="ignore")`
Added the following bipolar channels:
Fc5.-Fc3.

#### Result
If `on_bad` is "raise" or "warn", the bipolar channel ['Fc5.-Fc3.'] will be marked as bad.

### CODE REVIEW

**Comment by larsoner** [comment]
Test failures look real and indicative of some problem

**Comment by moritz-gerster** [comment]
We tried everything to get as many tests pass as possible but some are still failing. @larsoner do you have some tips for us?

**Comment by larsoner** [comment]
Locally if I `git merge upstream/main` at least `pytest mne/channels mne/io` passes. Can you click the "allow edits by maintainers"? Then I can push this update and CIs should run again

**Comment by moritz-gerster** [comment]
> Locally if I `git merge upstream/main` at least `pytest mne/channels mne/io` passes. Can you click the "allow edits by maintainers"? Then I can push this update and CIs should run again

@larsoner sure, but I think I checked this box already?
image

File: doc/changes/names.inc [hunk 1]
```diff
@@ -166,6 +166,8 @@
 
 .. _Guillaume Favelier: https://github.com/GuillaumeFavelier
 
+.. _Hakimeh Aslsardroud: https://www.researchgate.net/profile/Hakimeh-Pourakbari
```
Comment (larsoner) [inline doc/changes/names.inc:169]: Please double-check this link @moritz-gerster , it's more of a guess from my end

**Comment by moritz-gerster** [comment]
Looks good @larsoner! Is there anything else I should do?

File: doc/changes/names.inc (hunk 1 above)
Comment (moritz-gerster) [reply doc/changes/names.inc:169]: Looks good @larsoner! Is there anything else I should do?

**Review by agramfort — APPROVED** [review]

**Pull request merged**